### PR TITLE
Be more explicit about returning deleted actors or not

### PR DESCRIPTION
**Pull request by Nutomic**
Still have to test this.

### CODE REVIEW

File: crates/db_schema/src/impls/community.rs [hunk 1]
```diff
@@ -332,12 +332,22 @@ impl ApubActor for Community {
...
-      .filter(lower(name).eq(lower(community_name)))
-      .first::<Self>(conn)
+      .filter(lower(name).eq(lower(community_name)));
+    if include_deleted {
+      q.first::<Self>(conn)
+    } else {
+      q.filter(deleted.eq(false))
+        .filter(removed.eq(false))
+        .first::<Self>(conn)
+    }
```
Comment (Nutomic) [inline crates/db_schema/src/impls/community.rs:350]: This is a little awkward, not sure if theres a better way to write the query.

File: crates/apub/src/fetcher/mod.rs [hunk 2]
```diff
@@ -58,7 +58,7 @@ where
     let identifier = identifier.to_string();
     Ok(
       blocking(context.pool(), move |conn| {
-        DbActor::read_from_name(conn, &identifier)
+        DbActor::read_from_name(conn, &identifier, false)
```
Comment (Nutomic) [inline crates/apub/src/fetcher/mod.rs:61]: Search should probably not list deleted users or communities.

**Comment by Nutomic** [comment]
Works as expected.

File: crates/db_schema/src/impls/community.rs (hunk 1 above)
Comment (Nutomic) [reply crates/db_schema/src/impls/community.rs:350]: Like this? Gives me error "overflow evaluating the requirement `_: Sized`".
```

    let mut q = community
      .filter(local.eq(true))
      .filter(lower(name).eq(lower(community_name))).into_boxed();
    if !include_deleted {
      q = q.filter(deleted.eq(false))
        .filter(removed.eq(false)).into_boxed();
    }
    q.first::<Self>(conn)
```

Comment (dessalines) [reply crates/db_schema/src/impls/community.rs:350]: You do the into_boxed before filters I think.

**Comment by Nutomic** [comment]
updated

**Review by dessalines — APPROVED** [review]

**Pull request merged**